Database.py

```diff
@@ -0,0 +1,78 @@
+from datetime import datetime
+from xmlrpc.client import Boolean
+from Employee import INVALID_DATETIME, INVALID_STR, Employee
+from EmployeeContainer import EmployeeContainer, EmployeeAdmin, EmployeeOther, EmployeeSelf
+from pathlib import Path
+import csv
+from tkinter import messagebox
+
+
+class Database:
+    def __init__(self, initFPath: Path = Path("")) -> None:
+        self.employeeList: list[Employee] = list()
+        if initFPath == "":
+            return
+        elif initFPath.is_file():
+            with initFPath.open() as csvfile:
+                csvReader = csv.DictReader(csvfile)
+                for row in csvReader:
+                    # test what happens if there are different fields from Employee class
+                    self.employeeList.append(Employee(**row))
+        elif initFPath != "":
+            print("not a filepath\n")
+
+    # do we need to do any validity checks before adding?
+    def addEmployee(self, **kwargs) -> Boolean:
+        if len(self.search(Employee_ID=kwargs["Emp_ID"])) > 0:
+            return False
+        self.employeeList.append(Employee(**kwargs))
+        return True
+
+    def search(self, Department: str = INVALID_STR, fName: str = INVALID_STR, lName: str = INVALID_STR, Employee_ID: str = INVALID_STR, Title: str = INVALID_STR, oPhone: str = INVALID_STR, StartDate: datetime = INVALID_DATETIME, EndDate: datetime = INVALID_DATETIME) -> list:
+        foundList = list()
+        for emp in self.employeeList:
+            # when user sessions are implemented, this needs to be replaced with a proper switch
+            emp = EmployeeOther(emp)
+            if (Department == INVALID_STR or Department == emp.Dept) and (fName == INVALID_STR or fName == emp.getFName()) and (lName == INVALID_STR or lName == emp.getLName()) and (Employee_ID == INVALID_STR or Employee_ID == emp.EmpID) and (Title == INVALID_STR or Title == emp.Title) and (oPhone == INVALID_STR or oPhone == emp.OfficePhone) and (StartDate == INVALID_DATETIME or StartDate == emp.StartDate) and (EndDate == INVALID_DATETIME or EndDate == emp.EndDate):
+                foundList.append(emp)
+        return foundList
```

Comment:
we don't want that for all of them though, for employee ids, and can't really do that for the dates. that means that doing it with **kwargs would be more complicated.

Reply:
In my mind it would be much simpler in the way I mentioned it, we can have to switch cases for special kwargs like date, but just reading that if statement begs for a simpler solution

Reply:
that would also need a translator from each key to the variable name or rename the parameters to be the same as the var names. While doing it your way is doable and looks nicer, it would make this part of the code much more vulnerable to breaking when we change something else seemingly unrelated, cause many issues that vscode would have caught to instead show up as runtime errors instead, and make debugging awful.

Reply:
why would this break? I'm confused it would just be a for loop like this
for key, value in kwargs.items():
       ..... check based on key in database

Reply:
This is a more complete example
```
for emp in self.employeeList:
            emp = __employeeContainment(emp, usrAcc)
            emp_matches = True
            for key, value in kwargs:
                try:
                    if key is 'StartDate' or key is 'EndDate':
                        if value != emp[key]:
                            emp_matches = False
                            break
                        # compare dates
                    else:
                        # "in" checks for substrings
                        if not (value in emp[key]):
                            emp_matches = False
                            break
                except Exception:
                    print(f'Bad key in search database. Key: {key}')
                    emp_matches = False
                    break
            if emp_matches:
                foundList.append(emp)
                
```

We can still leave in the docstring that shows different kwargs that works, but then we don't need a giant if statement, or a huge list of kwargs

Reply:
key in kwargs isn't the same as attribute naames of emp. while they can be changed to be the same so that your code does work, relying on them to be the same as attribute names of a separate class is bad form and feels gross. if we renamed one of those attribute names in the future using the IDE, the IDE would not rename the parameter names. I'll restructure the if statement so it doesn't look as ugly.

Reply:
Ok, we can do that. I guess I just assumed that our attributes and column names would be the same. (I guess I'm used to using a tool such as SQLAlchemy, where when you set up your classes based on a database the fields have to match)